### PR TITLE
Prompt the user to create a ticket, if none exist, Resolves #14

### LINKED ISSUE #14: Initial landing home screen is blank , fix it by prompting a message to user to create ticket.

### PR DESCRIPTION
**Pull request by MadhavDhall**
### Related Issue  
Closes #14   

### Type of Change
- [ ] New Feature  
- [x] Bug Fix  
- [ ] Code Refactor  
- [ ] Documentation Update  
- [ ] Other (please specify): 

### Description of Change  
Added a card with similar style as of ticket which displays, "No Tickets Found" and clicking on the card it redirects to page to create new ticket. This will display only when no ticket exists.

### Implementation Details  
Added a card with similar style as of ticket which displays, "No Tickets Found" and clicking on the card it redirects to page to create new ticket. This will display only when no ticket exists ie when ticket array is empty.

### Demo  

https://github.com/user-attachments/assets/354ec0fe-e7f4-4de5-b486-f3270116d041

### CODE REVIEW

**Comment by MadhavDhall** [comment]
Kindly review this PR @AyushDharDubey

**Comment by himani-r5** [comment]
> Kindly review this PR @AyushDharDubey

Putting all the information in the card , that is , no tickets found and as well as create ticket now , feels a bit confusing for a first time user. To improve the user experience, show the message of no tickets found in the card but have a button for "create your first ticket now" so that they know what to click to get started. You can try any other suitable way to improve the user experience as well ,if you wish to.

**Comment by MadhavDhall** [comment]
Hi @himani-r5 mam
As said i have added a button to create ticket in the card itself.

https://github.com/user-attachments/assets/1866a1cb-71c0-4f13-bfb6-693c804124b7

Kindly resolve the issue by merging the PR.

File: app/page.jsx [hunk 1]
```diff
@@ -32,6 +33,7 @@ const Dashboard = () => {
 
     fetchTickets();
   }, []);
+  console.log(tickets.length);
```
Comment (AyushDharDubey) [inline app/page.jsx:36]: remove this line

File: app/(components)/NoTicketExists.jsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+const NoTicketExists = () => {
+    return (
+        <div className="lg:grid grid-cols-2 xl:grid-cols-4 gap-4">
+            <div
+                className="flex flex-col hover:bg-card-hover bg-card rounded-md shadow-lg p-3 m-2"
+            >
+                <h4 className="mb-1">No Tickets Found</h4>
+                <hr className="h-px border-0 bg-page mb-2 " />
+                <p className="whitespace-pre-wrap">There are currently no tickets created till now.</p>
```
Comment (AyushDharDubey) [inline app/(components)/NoTicketExists.jsx:11]: "There are no tickets available at the moment."

File: app/page.jsx [hunk 3]
```diff
@@ -62,6 +64,7 @@ const Dashboard = () => {
       </div>
 
       <div>
+        {tickets.length === 0 ? <NoTicketExists /> : ""}
```
Comment (AyushDharDubey) [inline app/page.jsx:66]: move the `getUniqueGroups()` call into the `else` block, so it only executes when tickets are present. The code should look like this:

```javascript
{tickets.length === 0 ? <NoTicketExists /> : 
getUniqueGroups().map((group, groupIndex) => (
  <div key={groupIndex} className="mb-4">
    <h2>{group}</h2>
  ....
```

**Comment by MadhavDhall** [comment]
Sir @AyushDharDubey 
I have made all the changes as said.
Kindly move on by merging the PR.

**Comment by AyushDharDubey** [comment]
@MadhavDhall i feel it would be better if the NoTicketExists component would be rendered even before the view by section
i.e the check 
```javascript
        {tickets.length === 0 ? <NoTicketExists /> :
```
instead of being at line 66 might better be at line 50



apart from this, please make sure that your commit messages clearly reflect your changes.
thank you

File: app/(components)/NoTicketExists.jsx
Comment (AyushDharDubey) [inline app/(components)/NoTicketExists.jsx:1]: please rename the file and it's default component into NoTicketExistsCard

**Comment by MadhavDhall** [comment]
Hi @AyushDharDubey sir
I have made all changes as suggested.
Kindly merge the PR.
Screenshot 2024-12-27 at 1 26 50 PM

File: app/page.jsx [hunk 4]
```diff
@@ -46,46 +47,49 @@ const Dashboard = () => {
...
+                    ))}
+                </div>
+              </div>
+            ))}
           </div>
-        ))}
-      </div>
+        </>}
     </div>
-  );
```
Comment (AyushDharDubey) [inline app/page.jsx:88]: I presume this one was unintentional

**Comment by AyushDharDubey** [comment]
> Kindly merge the PR.

@MadhavDhall, the maintainers are here to merge the PR only, and it will ultimately be merged once all necessary reviews are completed, but first we need to ensure we are satisfied with the changes before proceeding

File: app/(components)/NoTicketExistsCard.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    return (
+        <div className="lg:grid grid-cols-2 xl:grid-cols-4 gap-4">
+            <div
+                className="flex flex-col hover:bg-card-hover bg-card rounded-md shadow-lg p-3 m-2"
+            >
+                <h4 className="mb-1">No Tickets Found</h4>
+                <hr className="h-px border-0 bg-page mb-2 " />
+                <p className="whitespace-pre-wrap">There are no tickets available at the moment.</p>
+
+                <div className="flex-grow"></div>
```
Comment (AyushDharDubey) [inline app/(components)/NoTicketExistsCard.jsx:13]: is this intentional?

File: app/page.jsx (hunk 4 above)
Comment (MadhavDhall) [reply app/page.jsx:88]: Sorry sir, i am not able to understand what. do you want to say.

**Comment by AyushDharDubey** [comment]
The contents of the card might look better if aligned to the center.

@himani-r5 @MadhavDhall, what do you say?

File: app/page.jsx (hunk 4 above)
Comment (AyushDharDubey) [reply app/page.jsx:88]: i think you removed the semicolon at line no 92 by mistake?

Comment (MadhavDhall) [reply app/page.jsx:88]: Oh achha my bad 🥲. Resolved in latest commit.

File: app/(components)/NoTicketExistsCard.jsx (hunk 5 above)
Comment (MadhavDhall) [reply app/(components)/NoTicketExistsCard.jsx:13]: No actually, I copied the card code from ticket card where it was there. I tried removing it and it works completely fine. May i remove it and commit?

**Comment by MadhavDhall** [comment]
> The contents of the card might look better if aligned to the center.
> 
> @himani-r5 @MadhavDhall, what do you say?
> 
> i mean on wider displays the card looks this
> 
> image

Agreed sirji.

**Comment by himani-r5** [comment]
> The contents of the card might look better if aligned to the center.
> 
> @himani-r5 @MadhavDhall, what do you say?
> 
> i mean on wider displays the card looks this
> 
> image

Yes they would … having a better UI wont hurt 🥹

**Comment by MadhavDhall** [comment]
@AyushDharDubey @himani-r5, as said i have removed unnecessary line and aligned card's content at center. Anything else you feel?

**Comment by AyushDharDubey** [comment]
Good work @MadhavDhall,

the PR is looking good enough and is ready to be merged

**Pull request merged**